kafka-plugins-0.10/src/main/java/io/cdap/plugin/common/KafkaHelpers.java

```diff
@@ -120,4 +121,23 @@ public static void validateKerberosSetting(@Nullable String principal, @Nullable
       throw new IllegalArgumentException(message);
     }
   }
+
+  /**
+   * Validates whether the principal and keytab are both set or both of them are null/empty.
+   * Stores the result in the provided failureCollector.
+   *
+   * @param principal Kerberos principal
+   * @param keytab Kerberos keytab for the principal
+   * @param collector input failureCollector into which the error will be added if present
+   */
+  public static void validateKerberosSetting(@Nullable String principal, @Nullable String keytab,
+                                             FailureCollector collector) {
+    if (Strings.isNullOrEmpty(principal) != Strings.isNullOrEmpty(keytab)) {
+      String emptyField = Strings.isNullOrEmpty(principal) ? "principal" : "keytab";
+      String message = emptyField + " is empty. When Kerberos security is enabled for Kafka, then both the principal " +
```

Comment:
If the principal is defined but the keytab is not, this should be a simple message: "A keytab must be specified." 

If the keytab is defined by the principal is not, there should be a simple message: "A principal must be specified."

No need for a corrective action.